### DISCUSSION TITLE
isValid state doesn't changed in mui material Dialog

### DISCUSSION THREAD

**Discussion by ArtemEkzarho** [react-hook-form/react-hook-form#7663] [Show and tell] [answered]
Hi all, guys! I facing  strange behaviour. IsValid state doesn't change inside material <Dialog>. I am getting IsValid from useFormState hook.  What am i doing wrong? Could you please help me?

demo here
https://codesandbox.io/s/virtualize-material-demo-forked-gt7z7?file=/index.tsx

**Comment by Moshyfawn** [comment, marked as answer]
I believe, it's something got to do with the `<Dialog />` modal portal, that messes up the child components update on the form's state change in the Context.
I'd suggest to move your form initialization inside the `<Dialog />` with the rest of your form. See [this CSB](https://codesandbox.io/s/virtualize-material-demo-forked-jb50s?file=/index.tsx) for example

**Reply by ArtemEkzarho** [reply]
What if the form is needed outside the dialogue?)

**Reply by Moshyfawn** [reply]
Take a look at #6634
